utils/terminfo.ml

```diff
@@ -13,14 +13,35 @@
 (*                                                                        *)
 (**************************************************************************)
 
-(* Basic interface to the terminfo database *)
+open Printf
+
+external isatty : out_channel -> bool = "caml_sys_isatty"
+external terminfo_rows: out_channel -> int = "caml_terminfo_rows"
 
 type status =
   | Uninitialised
   | Bad_term
-  | Good_term of int
-;;
-external setup : out_channel -> status = "caml_terminfo_setup";;
-external backup : int -> unit = "caml_terminfo_backup";;
-external standout : bool -> unit = "caml_terminfo_standout";;
-external resume : int -> unit = "caml_terminfo_resume";;
+  | Good_term of int  (* number of lines of the terminal *)
+
+let setup () = 
+  let term = try Sys.getenv "TERM" with Not_found -> "" in
+  (* Same heuristics as in Misc.Color.should_enable_color *)
+  if term <> "" && term <> "dumb" && isatty stderr then begin
+    let rows = terminfo_rows stderr in
```

Comment:
Previously the `setup` analysis was done on `stdout` as per the given argument. You now do this on `stderr` is that change intended  ?

Reply:
The change was for consistency with `Misc.Color.should_enable_color`, which uses `stderr`.

However, a point could be made that the latter is oriented towards compiler error messages (on stderr) while the present code is targeted at the toplevel REPL error messages (on stdout, if I'm not mistaken).

I don't have a strong feeling either way.

Reply:
I'd say it's better invoke `isatty` and `terminfo_rows` on the fd that you are actually going to use for output and in this case it seems to be stdout:

```
> ocaml -noinit 2> /dev/null
        OCaml version 4.03.0

# let f x = let;;
Error: Syntax error
```

Reply:
OK for stdout.  See commit 88c2987.